classes/UpgradeSelfCheck.php

```diff
@@ -186,7 +187,8 @@ public function getWarnings(): array
     {
         $warnings = [
             self::MODULE_VERSION_IS_OUT_OF_DATE => !$this->isModuleVersionLatest(),
-            self::TEMPERED_FILES_LIST_NOT_EMPTY => !empty($this->getTamperedFiles()),
+            self::CORE_TEMPERED_FILES_LIST_NOT_EMPTY => !empty(array_merge($this->getCoreAlteredFiles(), $this->getCoreMissingFiles())),
+            self::THEME_TEMPERED_FILES_LIST_NOT_EMPTY => !empty(array_merge($this->getThemeAlteredFiles(), $this->getThemeMissingFiles())),
```

Comment:
In terms of performance and memory consumption, I wondering if checking the arrays without creating a new one would be better:

```suggestion
            self::CORE_TEMPERED_FILES_LIST_NOT_EMPTY => !empty($this->getCoreAlteredFiles()) || !empty($this->getCoreMissingFiles()),
            self::THEME_TEMPERED_FILES_LIST_NOT_EMPTY => !empty($this->getThemeAlteredFiles() || !empty($this->getThemeMissingFiles()),
```